provider/blob/src/export/blob_exporter.rs

```diff
@@ -29,6 +29,16 @@ pub struct BlobExporter<'w> {
     sink: Box<dyn std::io::Write + Sync + 'w>,
 }
 
+impl core::fmt::Debug for BlobExporter<'_> {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        f.debug_struct("BlobExporter")
+            .field("resources", &self.resources)
+            .field("unique_resources", &self.unique_resources)
+            .field("sink", &"???")
```

Comment:
ditto, use something that indicates omission rather than an error.

```suggestion
            .field("sink", &"<sink>")
```